articles/why-and-how-zk-snark-works/3.non_interactive_zero-knowledge_of_a_polynomial.md

```diff
@@ -0,0 +1,164 @@
+# Chapter 3
+
+## 3.1. **The Fundamental Theorem of Algebra**
+
+It states that any polynomial can be factored into linear polynomials as long as it is solvable. 
+
+Example:
+$$(x - a_{0})(x - a_{1})...(x - a_{n}) = 0$$
+
+Define *p(x)* as a sample polynomial, and *p(x)* is the multiplication of those cofactors *t(x)*.
+
+$$h(x) = \frac{p(x)}{t(x)}$$
+
+In particular:
+Given:
+$$p(x)  = x^3 - 3x^2 + 2x$$
+
+$$t(x) = (x-1)(x-2)$$
+
+The Verifier samples a random value 23, calculates  $$t = t(23) = (23 - 1)(23 - 2) = 462$$ and gives 23 to the Prover.
+
+The Prover calculates $h(x) = p(x)$, $t(x) = x$, evaluates $$p = p(23) = 10626$$ $$h = h(23) = 23$$ and provides p, h to the Verifier.
+
+The Verifier then checks that:  $$p = t \cdot h: 10626 = 462 \cdot 23$$ 
+This is true, and therefore the statement is proven.
+
+## 3.2. Modular Arithmetic 
+
+## 3.3. Encrypted Polynomial
+
+Let's define *E* : $$E(x) = g^{v} \pmod{n}$$
```

Comment:
E(x) = g^v ?

Reply:
@Draply resolved?

Reply:
oh! i forgot to commit it . *v* is monomial so it has x in